File: apps/api/models/organization.py
from typing import Optional

from pydantic import Field

from .base import BaseDocument


class Organization(BaseDocument):
    """Organization document model."""

    name: str = Field(..., description="Organization name")
    slug: str = Field(..., description="Organization URL slug (unique)")

    @classmethod
    def from_mongo(cls, data: dict) -> Optional["Organization"]:
        """Create Organization instance from MongoDB document."""
        if data is None:
            return None
        if "_id" in data:
            data["_id"] = str(data["_id"])
        return cls(**data)

File: apps/web/lib/queries/organization.ts
export const CREATE_ORGANIZATION = `
  mutation CreateOrganization($name: String!) {
    createOrganization(name: $name) {
      id
      name
      slug
    }
  }
`;

export type CreateOrganizationResponse = {
  createOrganization: {
    id: string;
    name: string;
    slug: string;
  };
};
